### PR TITLE
Remove influence of cdrom or other non-disk disks

### PR DESCRIPTION
**Pull request by chloerh**
Signed-off-by: Haijiao Zhao <haizhao@redhat.com>

### CODE REVIEW

**Comment by chunfuwen** [comment]
please attach test result after the change.

File: libvirt/tests/src/virtual_disks/virtual_disks_multivms.py [hunk 1]
```diff
@@ -127,6 +129,13 @@ def get_vm_disk_xml(dev_type, dev_name, **options):
     # Initialize VM list
     vms_list = []
     try:
+        # Remove cdrom/floppy or other non-disk disks
+        for vm_name in vm_names:
+            vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+            disk_devices = vmxml.get_devices('disk')
+            for disk in disk_devices:
+                if disk.device != 'disk':
+                    virsh.detach_disk(vm_name, disk.target['dev'], extra='--config', debug=True)
```
Comment (balamuruhans) [inline libvirt/tests/src/virtual_disks/virtual_disks_multivms.py]: can we check whether the detach_disk is succeeded else let us raise.

**Comment by dzhengfy** [comment]
@chloerh, please add some meaningful commit messages to state what problems your codes solve, thanks.

**Comment by iccaszhulili** [comment]
rh-caselink retest please

**Pull request closed without merging**